### PR TITLE
adding status as enum to entities

### PR DESCRIPTION
**Pull request by himanshugoyal1065**
Changing the status of the entites to enums.

### CODE REVIEW

File: spoutbreeze-agent/src/main/java/org/spoutbreeze/agent/services/AgentQueueListener.java [hunk 1]
```diff
@@ -79,4 +101,18 @@ public void startABroadcastSession(final BroadcastMessage broadcastMessage) {
...
+    /**
+     * update the broadcast status
+     * @param broadcastMessage the broadcast message
+     */
+    public void updateBroadcastStatus(final BroadcastMessage broadcastMessage) {
+        final Broadcast broadcast =
+                broadcastRepository.findById(broadcastMessage.getId())
+                        .orElseThrow(() -> new RuntimeException("no broadcast with id " + broadcastMessage.getId()));
+
+        broadcast.status = BroadcastStatus.LIVE;
```
Comment (himanshugoyal1065) [inline spoutbreeze-agent/src/main/java/org/spoutbreeze/agent/services/AgentQueueListener.java:114]: What's the difference between BroadcastStatus.LIVE and BroadcastStatus.ASSIGNED?

Comment (GhaziTriki) [reply spoutbreeze-agent/src/main/java/org/spoutbreeze/agent/services/AgentQueueListener.java:114]: The difference should be explained later in the docs.
ASSIGNED comes before LIVE. When the Broadcast gets into the manager; which is distribution system (load-balancer); it assigns a broadcast, now it becomes ASSIGNED. The agent gets that request, the broadcast becomes live when the selenoid session is created.

File: spoutbreeze-commons/src/main/java/org/spoutbreeze/commons/enums/AgentStatus.java [hunk 2]
```diff
@@ -19,5 +19,5 @@
 package org.spoutbreeze.commons.enums;
 
 public enum AgentStatus {
-    ENABLED, DISABLED
+    ENABLED, DISABLED, BROADCASTING
```
Comment (GhaziTriki) [inline spoutbreeze-commons/src/main/java/org/spoutbreeze/commons/enums/AgentStatus.java:22]: We don't need the `BROADCASTING` status for the agent, the reason is that we can find it from `broadcasts` table and we will need more config later for selenoid, an agent can example have 5 broadcasts.

**Review by GhaziTriki — COMMENTED** [review]
We don't need that status.

File: spoutbreeze-commons/src/main/java/org/spoutbreeze/commons/enums/AgentStatus.java (hunk 2 above)
Comment (himanshugoyal1065) [reply spoutbreeze-commons/src/main/java/org/spoutbreeze/commons/enums/AgentStatus.java:22]: Sure, makes sense.

**Review by GhaziTriki — APPROVED** [review]

**Pull request merged**